src/rpc/service_eth.go

```diff
@@ -273,3 +275,36 @@ var DefaultProcessTransaction = func(tx *txtypes.Transaction, ctx context.Contex
 
 	return submitTx, nil
 }
+
+func (p *Processor) MonitorBalance(ctx context.Context, delayInSeconds int) {
+	timer := time.NewTicker(time.Duration(delayInSeconds) * time.Second)
+
+	for {
+		select {
+		case <-ctx.Done():
+			utils.Logger.Info().Msg("Stopping because context is done.")
+			return
+
+		case <-timer.C:
+			balance, err := p.Client.BalanceAt(context.Background(), *p.SigningAddress, nil)
+			if err != nil {
+				utils.Logger.Err(err).Msg("Failed to get balance")
+				continue
+			}
+
+			// Convert balance from Wei to Ether
+			ethValue := new(big.Float).Quo(new(big.Float).SetInt(balance), big.NewFloat(1e18))
+			balanceInFloat, _ := ethValue.Float64()
+
+			metrics.MetricsERPCBalance.Set(balanceInFloat)
+		}
+	}
+}
+
+func returnError(status int, msg error) *EncodingError {
```

Comment:
You also need to use this function to count the errors returned when decrypting a transaction

Reply:
when doing that, could you please also format the error returned in line 272 (can't comment on that line on this review)